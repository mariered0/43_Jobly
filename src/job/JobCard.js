import React from "react";
import { Card, CardTitle, List } from "reactstrap";

const JobCard = ({ title, salary, equity }) => {
  return (
    <Card body className="my-2" style={{ width: "100%" }}>
      <CardTitle tag="h5">{title}</CardTitle>
      <List type="unstyled">
        <li>salary: {salary}</li>
        <li>equity: {equity}</li>
      </List>
    </Card>
  );
};

export default JobCard;
