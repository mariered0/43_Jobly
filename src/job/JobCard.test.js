import React from "react";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";
import JoblyApi from "../api/api";

import UserContext from "../user/UserContext";

const token = JoblyApi.token;

it("renders without crashing", () => {
  render(
        <UserContext.Provider value={{ token }}>
          <JobCard />
        </UserContext.Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
        <UserContext.Provider value={{ token }}>
          <JobCard />
        </UserContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
