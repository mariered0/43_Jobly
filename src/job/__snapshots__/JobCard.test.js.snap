// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-3 position-relative card card-body"
    style="width: 100%;"
  >
    <h5
      class="card-title"
    />
    <ul
      class="list-unstyled"
    >
      <li>
        salary: 
      </li>
      <li>
        equity: 
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
